include/libtorrent/peer_connection_handle.hpp

```diff
@@ -110,7 +110,7 @@ struct TORRENT_EXPORT peer_connection_handle
 	bool operator==(peer_connection_handle const& o) const
 	{ return !lt(m_connection, o.m_connection) && !lt(o.m_connection, m_connection); }
 	bool operator!=(peer_connection_handle const& o) const
-	{ return lt(m_connection, o.m_connection) || lt(o.m_connection, m_connection); }
```

Comment:
it's not obvious that this change improves anything, is it? If the compiler fails to inline all of this, it will probably make more calls to ``owner_before`` for instance